sote/main.lua

```diff
@@ -264,3 +273,21 @@ end
 function love.wheelmoved(x, y)
 	ui.on_wheelmoved(x, y)
 end
+
+function love.updateFullscreen()
```

Comment:
I dont believe this is a love2d callback function? The `love` table shouldnt be modified - we can declare it as a standalone function in the scope its needed

Reply:
Would moving the function to the global scope be enough or is there some specific scope/file it you would prefer it be in?

Reply:
Making a new file for it may be most reasonable. Avoid globals whenever possible

Reply:
What the the design philosophy behind the file hierarchy? Is there some section that I should place to put the update function definition?

Reply:
Some of these could be better but as they are right now:
```
Inside sote/ which stores the game to allow for convenient `love sote` launching of the game from repos root directory through a terminal:
coa/ - assets for coats of arms
cpml/ - math library, outside of the engine for reasons I don't remember anymore
data/ - "large" game assets and other non code stuff, should be split up and move out of the data directory. iirc wasn't split up to keep licensing info easier to read
engine/ - non GPL and external files
emblems/ - related to coas, different type of graphics
game/ - the game itself
icons/ - open source icons
music/ - open source music, too long to keep in RAM, streamed in
sfx/ - open source sounds, short, kept in RAM
```
Inside `sote/` there's a few "bootup" scripts for the game, there shouldn't be any files being added there and all new game could ought to be within `sote/game/`

As for directories inside `sote/game/`. they mostly divide it by game system.
`climate`, `ecology`, `economy`, etc, are all phases in game's simulation and as such get a directory.
Besides that, there's a `scenes` directory with each subdirectory within corresponding to a concept of a game scene in engines like Unity or Godot (in SotE scenes are basically a FSM).
The `map-modes` directory stores all logic related to coloring tiles in the game world. Not a game system but relatively standalone piece of game code.
`entities` is for "class definitions" of various data types stored on the game world.
`raws` is for static data definitions that are loaded before the game itself starts. Corresponds to `raws` in Dwarf Fortress or the `common` folder in games by Paradox Interactive. Things like "biomes", "races", "events", etc.
Lastly, `ai`, I think is quite self expanatory.

As for the fullscreen function, keeping it on the options object should do the trick